tests/cephfs/nfs/test_cephfs_nfs_transfer_file_bw_nfs_and_localfs.py

```diff
@@ -0,0 +1,260 @@
+        dd(clients[0], local_fs_path, file_name="dd_file_local", bs="100M", count=20)
+
+        log.info(
+            "File Operations : move data b/w nfs mounts and local filesystem, \
+                verify the transfers and check for data integrity"
+        )
+        filename_nfs = f"{clients[0].node.hostname}dd_file_nfs"
+        filename_local = f"{clients[0].node.hostname}dd_file_local"
+
+        nfs_file_path = Path(f"{nfs_mounting_dir}{filename_nfs}")
+        local_file_path = Path(f"{local_fs_path}{filename_local}")
+        nfs_file_copied_path = Path(f"{local_fs_path}{filename_nfs}")
+        local_file_copied_path = Path(f"{nfs_mounting_dir}{filename_local}")
+
+        # Transfer a file from NFS to local file system and local filesysten to NFS and verify successful transfer
+        client1.exec_command(sudo=True, cmd=f"cp {nfs_file_path} {local_fs_path}")
+        client1.exec_command(sudo=True, cmd=f"cp {local_file_path} {nfs_mounting_dir}")
+
+        # Verify successful transfer
+        output, err = client1.exec_command(
+            sudo=True, cmd=f"diff -qr {nfs_file_path} {nfs_file_copied_path}"
+        )
+        output, err = client1.exec_command(
+            sudo=True, cmd=f"diff -qr {local_file_path} {local_file_copied_path}"
+        )
+
+        # Verify Data Integrity using MD5  of both the NFS file and the local file
+        def calculate_md5(client, file_path):
+            return client.exec_command(
+                sudo=True, cmd=f"md5sum {file_path} | awk '{{print $1}}'"
+            )
+
+        nfs_hash = calculate_md5(client1, nfs_file_path)
+        local_hash = calculate_md5(client1, local_file_path)
+        if nfs_hash == local_hash:
+            print("File export and data integrity verification successful.")
+        else:
+            print("File export successful, but data integrity verification failed.")
+
+        log.info(
+            "File Operations : Verify the file attributes are retained when a file is moved out of NFS Share, \
+                        and also validate the same on a local file copied to NFS mount"
+        )
+
+        def compare_file_attributes(attr1, attr2):
+            return all(
+                attr1[key] == attr2[key]
+                for key in ["Permission", "Uid", "Gid", "Size", "Links"]
+            )
+
+        src_nfs_file_attributes = fs_util.get_stats(
+            clients[0], file_path=nfs_file_path, validate=True
+        )
+        des_nfs_file_attributes = fs_util.get_stats(
+            clients[0], file_path=nfs_file_copied_path, validate=True
+        )
+        if compare_file_attributes(src_nfs_file_attributes, des_nfs_file_attributes):
+            log.info(
+                "File Attributes on a NFS File are Retained after copying to local filesystem"
+            )
+        else:
+            log.info(
+                "File Attributes are not retained on a NFS File after copying to local filesystem"
+            )
+
```

Comment:
The next code is mostly repititive, can we add two set of src and dst file paths as array of arrays and iter with for each set, if possible?

Reply:
Done